smdebug/tensorflow/keras.py

```diff
@@ -584,14 +717,13 @@ def _on_any_batch_end(self, batch, mode, logs=None):
         if not is_tf_version_2x() or (is_tf_version_2x() and not tf.executing_eagerly()):
             self._remove_fetches_and_callbacks(mode)
 
+        self._save_tensors_post_step(batch, logs)
```

Comment:
Moved it to this position so that it is called before `self._prepare_non_layer_tensors()`.